### PR TITLE
Observe multiple nodes

### PR DESCRIPTION
**Pull request by mkacper**
This PR make it possible to observe/trace multiple nodes.

@michalslaski @mentels please let me know what do you think :)

### CODE REVIEW

File: apps/epl/src/epl.erl [hunk 1]
```diff
@@ -23,24 +27,75 @@
...
 
+%% Add calling process to every epl_tracers' subscribers list
+-spec subscribe() -> [ok].
 subscribe() ->
-    epl_tracer:subscribe().
+    Nodes = get_all_nodes(),
+    [subscribe(N) || N <- Nodes].
+
+%% Add calling process to Node tracer's subscribers list
+-spec subscribe(default_node) -> ok;
```
Comment (mentels) [inline apps/epl/src/epl.erl]: In the subscription functions you could use `node() | default_node` as spec for the `Node`.

File: apps/epl/src/epl.erl [hunk 2]
```diff
@@ -23,24 +27,75 @@
 %% ===================================================================
 %% API functions
 %% ===================================================================
+
+%% Lookup for given Key in epl_priv ets
+-spec lookup(Key :: term()) -> [tuple()].
 lookup(Key) ->
     ets:lookup(epl_priv, Key).
 
+%% Add calling process to every epl_tracers' subscribers list
```
Comment (mentels) [inline apps/epl/src/epl.erl]: Use the `Edoc` for documenting the public functions (http://erlang.org/doc/apps/edoc/chapter.html). For example the `@doc` tag and the `' for variables.

File: apps/epl/src/epl.erl [hunk 3]
```diff
@@ -23,24 +27,75 @@
...
+    unsubscribe(Node, self()).
+
+%% Remove provided Pid from Node tracer's subscribers list
+-spec unsubscribe(Node :: atom(), Pid :: pid()) -> ok.
+unsubscribe(Node, Pid) ->
+    epl_tracer:unsubscribe(Node, Pid).
 
-unsubscribe(Pid) ->
-    epl_tracer:unsubscribe(Pid).
+%% Run provided Fun with Args on every observed nodes
```
Comment (mentels) [inline apps/epl/src/epl.erl]: `on every of the observed nodes` or `all observed nodes`.

**Review by mentels — APPROVED** [review]

File: apps/epl/src/epl.erl [hunk 4]
```diff
@@ -23,24 +27,73 @@
 %% ===================================================================
 %% API functions
 %% ===================================================================
+
+%% @doc Lookups for given `Key' in epl_priv ets.
+-spec lookup(Key :: term()) -> [tuple()].
 lookup(Key) ->
     ets:lookup(epl_priv, Key).
 
+%% @doc Adds calling process to every epl_tracers' subscribers list.
```
Comment (michalslaski) [inline apps/epl/src/epl.erl]: For each node in the cluster we spawn a gen_server process, which is implemented in the epl_tracer module. Process calling this function is added to the subscribers list, which is kept in the state of the ep_tracer gen_server.

**Review by michalslaski — APPROVED** [review]

**Pull request merged**